drivers/spi/spi_nrfx_spim.c

```diff
@@ -310,6 +311,43 @@ static int init_spim(struct device *dev, const nrfx_spim_config_t *config)
 	return 0;
 }
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static int spim_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
+				void *context, device_pm_cb cb, void *arg)
+{
+	static u32_t current_state = DEVICE_PM_ACTIVE_STATE;
+	const struct spi_nrfx_config *dev_config = get_dev_config(dev);
+
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+		u32_t new_state = *((const u32_t *)context);
+
+		if (new_state != current_state) {
+			if (new_state == DEVICE_PM_ACTIVE_STATE) {
+				nrfx_spim_init(&dev_config->spim,
```

Comment:
We should use consistent approach in these drivers, i.e. use either `nrfx_spi[m]_init` or `init_spi[m]` in both. I think the latter option will be better as it will be also consistent with what we'll have in TWI/TWIM drivers.